### PR TITLE
Fix homepage css

### LINKED ISSUE #2023: Documentation Issue: Contributing section on home page has alignment issues
## Bug / Documentation: 

TL;DR: The `Contributing` heading on the home page is ugly in full screen and half-screen views

### Replication steps

1. Check out the repo
2. `pip install -Ie .[dev]`
3. `./make.py serve`
4. Go to http://localhost:8000
5. Scroll down

####  Full screen
image

#### Half screen
image

### PR DESCRIPTION
**Pull request by lbcai**
Fixes #2023 

- Adjusted the top header where the sprinter and the star button are
- Made the Contributing section always have 1 column since there are only 2 items there. But if 2 columns are preferred, I'll change that back
- Added a little margin between sections
- Made Social and Learning Resources section swap to 1 column if screen is small enough, just like the other sections
- Adjusted star button for dark mode users. Before:
image
After:
image
It's not currently possible to make the iframe background match the color of the dark mode docs. To fix that, we could use this other solution (https://buttons.github.io/) that involves loading a Javascript script, but it seems like this background issue is only present in Firefox and not in Chrome so I imagine it's low priority.

### CODE REVIEW

File: doc/_static/css/custom.css [hunk 1]
```diff
@@ -7,15 +7,64 @@ body {
     .toctree-wrapper > ul {
         columns: 2;
     }
+    .toc-outside-links {
+        columns: 2;
+    }
+}
+.item-framing-box {
```
Comment (pushfoo) [inline doc/_static/css/custom.css]: 1. Can we add a brief comment above this group of rules to explain what they're doing and why? It could also link the issue / PR to avoid having the extra step of git blame.
2. Is there a more specific prefix we can use than `item-framing-`?

File: doc/_static/css/custom.css [hunk 2]
```diff
@@ -7,15 +7,64 @@ body {
...
+    align-items: flex-start;
+    display: flex;
+    margin: 10px;
+}
+.item-framing-box .single-col-box {
```
Comment (pushfoo) [inline doc/_static/css/custom.css]: I'm not yet sure on this approach. My first reaction is to use child rules rather than applying to all descendants, but I don't yet fully understand your intent.

**Review by pushfoo — CHANGES_REQUESTED** [review]
TL;DR: Is it possible to split the GitHub stars fix into a separate, smaller PR to merge it in faster?

Visually, this definitely seems to work and I like it. However:
* We're mixing multiple concerns right now
* Adding some clarification comments might help us in the future
* Some of the CSS class names are a little vague, but that might be OK with comments

File: doc/_static/css/custom.css (hunk 2 above)
Comment (lbcai) [reply doc/_static/css/custom.css]: You're right, nice tip. I swapped them to child selectors on seeing that child selectors are more efficient than descendant selectors and less likely to cause someone else issues down the line.

File: doc/_static/css/custom.css (hunk 1 above)
Comment (lbcai) [reply doc/_static/css/custom.css]: I renamed some things, moved some things around to group them more nicely, and added comments. Hope it's more understandable now. Let me know if I should make some more changes. There isn't a github star button change in this PR, I just edited the size and position for visual purposes.

**Review by pvcraven — APPROVED** [review]

**Comment by pvcraven** [comment]
The PR makes things better. Bit more code churn than I'd like.

Looking at it, I think that whole toc-outside-links can be removed. It was originally created when I didn't know how to do links (or was it supported at that point?) in a TOC, so I faked one.

See: [this other PR](https://github.com/pythonarcade/arcade/pull/2037)

Think this PR should be merged, then go back and simplify.
